livekit/src/rtc_engine/mod.rs

```diff
@@ -375,7 +451,7 @@ impl EngineInner {
     ) -> EngineResult<()> {
         let mut running_handle = self.running_handle.write().await;
         if running_handle.is_some() {
-            unreachable!("engine is already connected");
+            panic!("engine is already connected");
```

Comment:
is this the common way of error handling in Rust? would this crash the program?

Reply:
Yes, it's more like an assert here, this should never happens, gonna change the panic to assert